api/core/providers/openai_image/openai_image_provider.py

```diff
@@ -0,0 +1,168 @@
+        return True
+
+    @override
+    async def _extract_and_log_rate_limits(self, response: Response, options: ProviderOptions):
+        await self._log_rate_limit_remaining(
+            "requests",
+            remaining=response.headers.get("x-ratelimit-remaining-requests"),
+            total=response.headers.get("x-ratelimit-limit-requests"),
+            options=options,
+        )
+        await self._log_rate_limit_remaining(
+            "tokens",
+            remaining=response.headers.get("x-ratelimit-remaining-tokens"),
+            total=response.headers.get("x-ratelimit-limit-tokens"),
+            options=options,
+        )
+
+    @override
+    def _compute_prompt_token_count(
+        self,
+        messages: list[dict[str, Any]],
+        model: Model,
+    ) -> float:
+        raise NotImplementedError("_compute_prompt_token_count should not be called for OpenAI Image")
+
+    @override
+    @classmethod
+    def _compute_completion_token_count(
+        cls,
+        response: str,
+        model: Model,
+    ) -> int:
+        raise NotImplementedError("_compute_completion_token_count should not be called for OpenAI Image")
+
+    # TODO: deprecate this method
+    @override
+    def _compute_prompt_image_count(
+        self,
+        messages: list[dict[str, Any]],
+    ) -> int:
+        return 0
+
+    @override
+    async def _compute_prompt_audio_token_count(
+        self,
+        messages: list[dict[str, Any]],
+    ) -> tuple[float, float | None]:
+        return 0, None
+
+    @override
+    @classmethod
+    def standardize_messages(cls, messages: list[dict[str, Any]]) -> list[StandardMessage]:
+        return []
+
+    @override
+    async def _prepare_completion(
+        self,
+        messages: list[Message],
+        options: ProviderOptions,
+        stream: bool,
+    ) -> tuple[OpenAIImageRequest, LLMCompletion]:
+        prompt = "\n".join([m.content for m in messages])
+
+        raw = LLMCompletion(
+            messages=[{"role": "user", "content": prompt}],
+            usage=self._initial_usage(messages),
+            provider=self.name(),
+        )
+        req = OpenAIImageRequest(prompt=prompt, n=1, image=None, mask=None, model=options.model)
+
+        return req, raw
+
+    @override
+    async def _execute_request(self, request: OpenAIImageRequest, options: ProviderOptions) -> Response:
+        url = (
+            "https://api.openai.com/v1/images/edits"
+            if request.is_edit_request
+            else "https://api.openai.com/v1/images/generations"
+        )
+
+        async with self._open_client(url) as client:
+            return await client.post(
+                url,
+                # Probably have to send the mask and image as files
+                headers={"Authorization": f"Bearer {self._config.api_key}"},
+                json=request.model_dump(mode="json", exclude_none=True, by_alias=True),
+                timeout=options.timeout,
```

Comment:
@guillaq which timeout value is used by default ? It should be higher for this provider.

Reply:
3min